django/db/backends/postgresql/operations.py

```diff
@@ -24,6 +25,7 @@ def get_json_dumps(encoder):
 
 
 class DatabaseOperations(BaseDatabaseOperations):
+    compiler_module = "django.db.backends.postgresql.compiler"
```

Comment:
The compiler approach was used over a set of feature flags because this optimization is very Postgres specific (`unnest`, native array support) and felt more alike to [the work to support MySQL'isms](https://github.com/django/django/blob/857b1048d53ebf5fc5581c110e85c212b81ca83a/django/db/backends/mysql/compiler.py#L6-L84).